first_year_scraper/excel_formatter.py

```diff
@@ -0,0 +1,137 @@
+from openpyxl import Workbook
+from openpyxl import load_workbook
+import slot
+import json
+
+
+def getkey(a, b):
+    '''returns key for cell[a][b] in excel'''
+    return str(chr(ord('A')+a-1))+str(b)
+
+# copy same values in merged cells
+def sanitise_sheets(valid_sheets, wb):
+    for sheet in valid_sheets:
+        ws = wb.get_sheet_by_name(sheet)
+        ws.delete_cols(1,2)
+        ws.delete_rows(1,2)
+        # merged_ranges = ws.merged_cell_ranges # get all merged ranges in sheet #TAKEN CARE OF BY PARSER
+        # for merged_range in merged_ranges:
+        #     boundary = merged_range.bounds
+        #     original_value = ws[getkey(boundary[0], boundary[1])].value
+        #     ws.unmerge_cells(str(merged_range))
+        #     for i in range(boundary[0], boundary[2]+1):
+        #         for j in range(boundary[1], boundary[3]+1):
+        #             ws[getkey(i,j)].value = original_value
+    # for sheet in valid_sheets:
+    #     ws = wb.get_sheet_by_name(sheet)
+    #     depth = 10
+    #     while(ws['A1'].value != "Period" and depth):
+    #         ws.delete_rows(1)
+    #         depth-=1;
+
+
+def format_cell(original_value):
+    '''converts cell text to actually usable array of subject name and room numbers'''
+    if(original_value is None):
+        pass
+    elif("physics" in original_value.lower()):
+        original_value = "phy_lab"
+    elif("drawing" in original_value.lower()):
+        original_value = "ed_lab"
+    elif("data" in original_value.lower()):
+        original_value = "pds_lab"
+    elif("electrical" in original_value.lower()):
+        original_value = "et_lab"
+    elif("process" in original_value.lower()):
+        original_value = "mech_lab"
+    elif("language" in original_value.lower()):
+        original_value = "hs_lab"
+    elif("chemistry" in original_value.lower()):
+        original_value = "chem_lab"
+    elif("l u n c h" in original_value.lower()):
+        original_value = "lunch_break"
```

Comment:
Yo Akshat, can you please help me understand why we are not using string comparison here? Is it because you might get a string like "Physics1" in `original_value` and you only care about the keyword "physics"?

Reply:
HAHA! I tried doing that but all thanks to ERP they misspelled PROGRAMMING to PROGAMMING in few timetables. So I just included the words I think they will keep the spelling right for in the future and are unique to the specific block. xD

Reply:
Oh! Cool. That makes the script more flexible.